cmd/workflow-controller/main.go

```diff
@@ -136,7 +136,7 @@ func NewRootCommand() *cobra.Command {
 				// for controlling the dummy metrics server
 				dummyCtx, dummyCancel := context.WithCancel(context.Background())
 				defer dummyCancel()
-				go wfController.RunPrometheusServer(dummyCtx, true)
+				wg := wfController.RunPrometheusServer(dummyCtx, true)
```

Comment:
This would be simpler than breaking encapsulation, no?
```suggestion
				var wg gosync.WaitGroup
				wg.Add(1)
				go func() {
					wfController.RunPrometheusServer(dummyCtx, true)
					wg.Done()
				}
```

Reply:
Yep, that looks nicer. Done in 5dbe063d15a3dc6bf66ccd5151a55eff042c0db1. Also, fixed a bug in `OnStoppedLeading` in e3f666c59c55f3ea6f1d50eeeaa4b8e89ebfa8bc